src/lib/date-tryparse-internal.jl

```diff
@@ -108,17 +108,18 @@ Returns a 3-element tuple `(values, pos, num_parsed)`:
             quote
                 pos > len && @goto done
                 $nullable, next_pos = tryparsenext(directives[$i], str, pos, len, locale)
-                isnull($nullable) && @goto error
-                $name = unsafe_get($nullable)
+                $nullable===nothing && @goto error
+                $name = something($nullable)
```

Comment:
Why do we need the `something` here?

Reply:
Ah, yes, you are right. Somehow I got confused here. Now fixed.